Prompt security litellm

lior-ps opened the pull request:
## Title

Add Prompt Security Guardrail

## Relevant issues

Add Prompt Security Guardrail

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ X ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ X ] I have added a screenshot of my new test passing locally 
- [  ] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [ X ] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🆕 New Feature
📖 Documentation
✅ Test

## Changes

lior-ps committed:
add prompt security guardrails provider

lior-ps committed:
cosmetic

lior-ps committed:
small

tabac-ps committed:
add file sanitization and update context window

tabac-ps committed:
add pdf and OOXML files support

tabac-ps committed:
add system prompt support

tabac-ps committed:
add tests and documentation

lior-ps committed:
Merge remote-tracking branch 'upstream/main' into prompt_security_litellm

lior-ps committed:
remove print

lior-ps committed:
fix PLR0915 Too many statements (96 > 50)

lior-ps committed:
cosmetic

lior-ps committed:
fix mypy error

krrishdholakia commented:
@lior-ps can you fix the conflicts? I'm unable to merge this 
<img width="1143" height="633" alt="Screenshot 2025-11-07 at 7 08 09 PM" src="https://github.com/user-attachments/assets/d42f8e38-10a4-4886-a1c7-18ed1a803922" />

krrishdholakia commented on tests/guardrails_tests/test_prompt_security_guardrails.py:
move inside `test_litellm/` so it runs on the github action

vitaly-ps committed:
Fix failed tests due to naming conflict of responses directory with same-named pip package

vitaly-ps committed:
Merge branch 'main' into prompt_security_litellm

vitaly-ps committed:
Fix mypy error: use 'aembedding' instead of 'embeddings' for async embedding call type

vitaly-ps committed:
Fix: Install enterprise package into Poetry virtualenv for tests

The GitHub Actions workflow was installing litellm-enterprise to system Python
using 'python -m pip install -e .', but tests run in Poetry's virtualenv using
'poetry run pytest'. This caused ImportError for enterprise package types.

Changed to 'poetry run pip install -e .' so the package is available in the
same virtualenv where pytest executes.

Fixes enterprise test collection errors in GitHub Actions CI.

lior-ps commented:
> @lior-ps can you fix the conflicts? I'm unable to merge this <img alt="Screenshot 2025-11-07 at 7 08 09 PM" width="1143" height="633" src="https://private-user-images.githubusercontent.com/17561003/511616922-d42f8e38-10a4-4886-a1c7-18ed1a803922.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3MTA1NDUsIm5iZiI6MTc2MjcxMDI0NSwicGF0aCI6Ii8xNzU2MTAwMy81MTE2MTY5MjItZDQyZjhlMzgtMTBhNC00ODg2LWExYzctMThlZDFhODAzOTIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA5VDE3NDQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhNTRmMDBjZDFlNTYxYjdiZTZjNmNkY2Q2Mzk0YTI5ZTY2YWM1NTNjN2NhNjEwM2U4NWU0YTM4MDJmMWZmNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.a__TXOvo2Q7osD0ABZjjtV_sgSyOqCHue4TzqfggZa4">

@krrishdholakia fixed the tests related to our change. the other failures are not related to our code. can you please have a look?

vitaly-ps committed:
Move Prompt Security guardrail tests to tests/test_litellm/

Per reviewer feedback, move test_prompt_security_guardrails.py from
tests/guardrails_tests/ to tests/test_litellm/proxy/guardrails/ so
it will be executed by GitHub Actions workflow test-litellm.yml.

This ensures the Prompt Security integration tests run in CI.

tabac-ps commented:
Hello @krrishdholakia ,
We would really appreciate this to be expedite. 
We have several users wishing to try the new guardrails and would like to stay on the main branch as there are new updates regularly. 
Is there something we can do in our side to push this forward?
Thank you!
